### ISSUE TITLE
.go directory causes error during release

### ISSUE THREAD

**Issue by sirlatrom** [goreleaser/goreleaser#1897]
**Describe the bug**

If there exists a directory `.go` in the working directory, it causes an error:

```
   ⨯ release failed after 17.83s error=failed to parse dir: .: read .go: is a directory
```

It should be possible for goreleaser to filter out such a directory. It is sometimes used in CI/CD jobs as a cache-managed GOPATH.

**To Reproduce**

Any config file.

> Make sure your config file is valid by running
> `goreleaser check -f path-to-config-file`.

**Expected behavior**

No errors.

**Environment (please complete the following information):**

- OS: linux
- OS version: Linux lx64pc0265 5.7.1-050701-generic #202006071230 SMP Sun Jun 7 12:32:55 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
- GoReleaser:
```
goreleaser version 0.147.0
commit: c39c8208f094ab854e4e7c8d70d8b74fef85fcb5
built at: 2020-11-12T13:53:56Z
built by: goreleaser
```
- GitLab version: 13.5.3-ee (b9d194b6b91)

**Additional context**
It's possible to work around if you have access to modify your GOPATH, but in our case not really.

**Comment by caarlos0** [comment]
Can confirm, here's a reproducible:

[asd.tar.gz](https://github.com/goreleaser/goreleaser/files/5536793/asd.tar.gz)

**Comment by caarlos0** [comment]
PS: this has been fixed upstream on go 1.16... will revert this change once 1.16 lands.
